jopa-impl/src/main/java/cz/cvut/kbss/jopa/model/metamodel/AbstractIdentifiableType.java

```diff
@@ -50,10 +59,18 @@ void addDeclaredQueryAttribute(final String name, final AbstractQueryAttribute<X
         declaredQueryAttributes.put(name, a);
     }
 
-    void setSupertype(AbstractIdentifiableType<? super X> supertype) {
-        assert supertype != null;
-        this.supertype = supertype;
-        supertype.addSubtype(this);
+    private void setClassSupertype(AbstractIdentifiableType<? super X> classSupertype) {
+        this.classSupertype = classSupertype;
+    }
+
+    void setSupertypes(Set<AbstractIdentifiableType<? super X>> supertypes) {
```

Comment:
Could you add explanatory comments here, as this method now has multiple side effects despite looking like a plain setter?

Reply:
Could you please check if added comment is sufficient ?

Reply:
I also meant the side effect of setting `classSupertype` for the non-interface class passed as argument.

Reply:
I deliberately did not mention it, as I considered it as an "implementation detail" with no function outside this class. 
It is purely a field contained and used in this class, and as such it takes no effect on caller. 

 
Of course, you have the final word, but for me, mentioning it makes no sense, since the caller cannot ever directly use this field.

Reply:
Ah, I see, `classSupertype` is not accessed anywhere outside of this class. In that case, it's ok, leave it without mentioning it in the javadoc. Just remove the setter for the field, I don't see any reason for it existing, as it is called only once.

Reply:
Great, the setter was removed. Thanks !